langchain/client/runner_utils.py

```diff
@@ -372,3 +375,107 @@ def run_on_examples(
             print(f"{i+1} processed", flush=True, end="\r")
     results[str(example.id)] = result
     return results
+
+
+def _get_session_name(
+    session_name: Optional[str],
+    llm_or_chain_factory: MODEL_OR_CHAIN_FACTORY,
+    dataset_name: str,
+) -> str:
+    if session_name is not None:
+        return session_name
+    current_time = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+    if isinstance(llm_or_chain_factory, BaseLanguageModel):
+        model_name = llm_or_chain_factory.__class__.__name__
+    else:
+        model_name = llm_or_chain_factory().__class__.__name__
+    return f"{dataset_name}-{model_name}-{current_time}"
+
+
+async def arun_on_dataset(
```

Comment:
Not a 1-way door, but current PR just exposes these as functions rather than extending the Client. Choice is whether the syntactic sugar of `client.run_on_dataset()` instead of `run_on_dataset(optional client)` is more useful than it is confusing (confusing here in that the client has diff methods depending on how you import)

Reply:
I think either is fine tbh